src/components/DesktopPatientTable/index.js

```diff
@@ -131,26 +137,34 @@ const DesktopPatientTable = (props) => {
 
   const loadMore = useCallback(() => {
     if (hasNext && isScrolledToEnd()) {
+      setIsShowLoadMoreSpinner(true);
       incrementPage();
     }
   }, [hasNext, selectedCases, page, searchText]);
 
   const isScrolledToEnd = () => {
     return (
-      wrapperRef.current.scrollTop >
-      wrapperRef.current.scrollHeight - wrapperRef.current.offsetHeight
+      wrapperRef?.current?.scrollTop >
+      wrapperRef?.current?.scrollHeight - wrapperRef?.current?.offsetHeight
     );
   };
 
   useEffect(() => {
     if (wrapperRef?.current) {
       wrapperRef.current.addEventListener('scroll', loadMore);
       return () => {
-        wrapperRef.current.removeEventListener('scroll', loadMore);
+        wrapperRef.current &&
+          wrapperRef.current.removeEventListener('scroll', loadMore);
       };
     }
   }, [wrapperRef?.current, loadMore]);
 
+  useEffect(() => {
+    if (!isShowSpinner) {
+      setIsShowLoadMoreSpinner(false);
+    }
+  }, [isShowSpinner]);
```

Comment:
any reason why we are creating a new variable and not just using `isShowSpinner`?